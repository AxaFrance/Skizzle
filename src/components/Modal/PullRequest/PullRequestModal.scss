/**
 * Pull request
 */

h2 {
	text-align: center;
	margin-bottom: 1rem;
}

.skz-PrModal {
	width: 100%;
	position: relative;
	overflow: auto;
	padding: 1rem;
	background-color: #f5f5f5;

	&__link {
		&:hover {
			opacity: 0.8;
			cursor: pointer;
		}
	}

	&__title {
		margin-bottom: 0.8rem;
		font-size: 1.5rem;
	}

	&-section {
		display: flex;
		flex-direction: column;

		&-comments {
			display: flex;
			padding: 0.5rem;
			cursor: pointer;
			background-color: #fff;
			border-radius: 4px;
			box-shadow: 0 2px 5px rgba(100, 100, 100, 0.1);
			transition: opacity linear 0.2s;

			&:hover {
				opacity: 0.8;
			}

			&__avatar {
				flex: 0 0 auto;
				width: 4rem;
				height: 4rem;
				margin-top: -1rem;
				margin-right: 0.5rem;
				border-radius: 50%;
				border: 2px solid #fff;
				background-color: #fff;
			}

			&__details {
				flex: 1 0 auto;
				width: calc(100% - 4.5rem);
			}

			&__author {
				display: flex;
				margin-bottom: 0.5rem;
				font-size: 0.8rem;
				font-weight: normal;
				color: #aaa;

				span {
					margin-left: auto;
				}
			}

			&__repo {
				margin-bottom: 0.5rem;
				font-size: 0.8rem;
			}
		}

		&__separator {
			border-left: 0.2rem solid #aaa;
			height: 3rem;
			margin-left: 2.5rem;

			&--hide {
				opacity: 0;
				margin-bottom: 2rem;
			}
		}
	}
}

@media (prefers-color-scheme: dark) {
	.skz-PrModal {
		background-color: #333;

		&__title {
			color: #fff;
		}

		&-section-comments {
			background-color: #444;

			&__avatar {
				border-color: #444;
				background-color: #444;
			}

			&__repo {
				color: #ccc;
			}
		}
	}
}
