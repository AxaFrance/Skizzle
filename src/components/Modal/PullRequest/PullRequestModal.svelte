<script>
  import Modal from '../Modal.svelte';
  import Loader from '../../Loader';
    const { shell } = require('electron');

  import {
    getTime,
    getFile,
    getAvatarUrl,
    updateCommentsWithUsername,
  } from './PullRequestModal.utils';

  import { othersProfile } from '../../../shared/store';

  export let commentsGroup;
  export let organizationName;
  export let pullRequest;

    const openUrl = () => shell.openExternal(pullRequest.url);

</script>

<style src="./PullRequestModal.scss">

</style>

<Modal>
<div slot="header" >
  <h1 class="skz-PrModal__title">{pullRequest.title}</h1>
  <p class="skz-PrModal__link"on:click={openUrl}>Voir sur Azure</p>
</div>
  <div class="skz-PrModal">
    {#await updateCommentsWithUsername(commentsGroup, othersProfile)}
      <Loader />
    {:then}
      {#each commentsGroup as commentGroup}
        <h2 class="skz-PrModal__title">{getFile(commentGroup)}</h2>
        {#each commentGroup.comments as comment, index}
          <div class="skz-PrModal-section">
            <div class="skz-PrModal-section-comments">
              <div class="skz-PrModal-section-comments__avatar">
                {#await getAvatarUrl(comment.author, organizationName)}
                  <img
                    class="skz-PrModal-section-comments__avatar"
                    alt={comment.author.displayName}
                    src="./assets/user.svg" />
                {:then avatar}
                  <img
                    class="skz-PrModal-section-comments__avatar"
                    alt={comment.author.displayName}
                    src="data:image/jpeg;base64,{avatar.value}" />
                {:catch error}
                  <img
                    class="skz-PrModal-section-comments__avatar"
                    alt={comment.author.displayName}
                    src="./assets/user.svg" />
                {/await}
              </div>
              <div class="skz-PrModal-section-comments__details">
                <h2 class="skz-PrModal-section-comments__author">
                  {comment.author.displayName} - {getTime(comment.lastUpdatedDate)}
                </h2>
                <p class="skz-PrModal-section-comments__repo">
                  {comment.content}
                </p>
              </div>

            </div>
            <div
              class={`skz-PrModal-section__separator${commentGroup.comments.length === 0 || index === commentGroup.comments.length - 1 ? '--hide' : ''}`} />
          </div>
        {/each}
        {:else}
          <p class="skz-PrModal__title">Pas de commentaire disponible sur cette pull request</p>
      {/each}
    {/await}
  </div>
</Modal>
