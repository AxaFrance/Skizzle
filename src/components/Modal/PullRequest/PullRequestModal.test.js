import { render } from '@testing-library/svelte';
import PullRequestModal from './PullRequestModal.svelte';

describe('PullRequestModal Component', () => {
	it('should render', () => {
		const { container } = render(PullRequestModal, {
			pullRequest: {
				title: 'PR',
			},
			commentsGroup: [
				{
					threadContext: {
						filePath: 'Modal.svelte',
					},
					comments: [
						{
							lastUpdatedDate: new Date(),
							author: {
								displayName: 'Moignont Christophe',
							},
							content: 'En attente de Pair test...',
						},
					],
				},
			],
		});

		expect(container).toMatchSnapshot();
	});
});
