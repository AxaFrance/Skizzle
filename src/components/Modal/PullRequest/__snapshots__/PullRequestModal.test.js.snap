// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PullRequestModal Component should render 1`] = `
<body>
  <div>
    <div
      class="skz-modal svelte-10r7fla"
    >
      <div
        class="skz-modal-container svelte-10r7fla"
      >
        <div
          class="skz-modal-header skz-modal-header--show svelte-10r7fla"
        >
          <div
            slot="header"
          >
            <h1
              class="skz-PrModal__title svelte-1o3z7ds"
            >
              PR
            </h1>
             
            <p
              class="skz-PrModal__link svelte-1o3z7ds"
            >
              Voir sur Azure
            </p>
          </div>
        </div>
         
        <div
          class="skz-modal-body svelte-10r7fla"
        >
           
          <div
            class="skz-PrModal svelte-1o3z7ds"
          >
            <div
              class="skz-loader  svelte-18xowzd"
            >
              <div
                class="skz-spinner  svelte-18xowzd"
              />
            </div>
          </div>
        </div>
         
        <div
          class="skz-modal-footer  svelte-10r7fla"
        />
      </div>
    </div>
  </div>
</body>
`;
